src/network_analysis.jl

```diff
@@ -192,6 +192,122 @@ function complexstoichmat(::Type{Matrix{Int}}, rn::ReactionSystem, rcs)
+    K = if sparse
+        fluxmat(SparseMatrixCSC{mtype, Int}, rcmap, rates)
+    else
+        fluxmat(Matrix{mtype}, rcmap, rates)
+    end
+end
+
+function fluxmat(::Type{SparseMatrixCSC{T, Int}}, rcmap, rates) where T
+    Is = Int[]
+    Js = Int[]
+    Vs = T[]
+    for (i, (complex, rxs)) in enumerate(rcmap)
+        for (rx, dir) in rxs
+            dir == -1 && begin
+                push!(Is, rx)
+                push!(Js, i)
+                push!(Vs, rates[rx])
+            end
+        end
+    end
+    Z = sparse(Is, Js, Vs, length(rates), length(rcmap))
+end
+
+function fluxmat(::Type{Matrix{T}}, rcmap, rates) where T
+    nr = length(rates); nc = length(rcmap)
+    K = zeros(T, nr, nc)
+    for (i, (complex, rxs)) in enumerate(rcmap)
+        for (rx, dir) in rxs
+            dir == -1 && (K[rx, i] = rates[rx])
+        end
+    end
+    K
+end
+
+function fluxmat(rn::ReactionSystem, pmap::Vector; sparse = false) 
+    pdict = Dict(pmap)
+    fluxmat(rn, pdict; sparse)
+end
+
+function fluxmat(rn::ReactionSystem, pmap::Tuple; sparse = false) 
+    pdict = Dict(pmap)
+    fluxmat(rn, pdict; sparse)
+end
+
+"""
+    massactionvector(rn::ReactionSystem, scmap = Dict())
+
+    Return the vector whose entries correspond to the "mass action products" of each complex. For example, given the complex A + B, the corresponding entry of the vector would be A*B, and for the complex 2X + Y, the corresponding entry would be X^2*Y. The ODE system of a chemical reaction network can be factorized as ``\frac{dx}{dt} = Y A_k Φ(x)``, where Y is the [`complexstoichmat`](@ref) and A_k is the negative of the [`networklaplacian`](@ref). This utility returns Φ(x).
+    Returns a symbolic vector by default, but will return a numerical vector if species concentrations are specified as a tuple, vector, or dictionary via scmap.
+"""
+function massactionvector(rn::ReactionSystem, scmap::Dict = Dict()) 
+    r = numreactions(rn); rxs = reactions(rn)
+    sm = speciesmap(rn); specs = species(rn)
+
+    if !all(r -> ismassaction(r, rn), rxs)
+        error("The supplied ReactionSystem has reactions that are not ismassaction. The mass action vector is only defined for pure mass action networks.")
+    end
+
+    if !isempty(scmap)
+        length(scmap) != length(specs) && error("Incorrect number of species concentrations were specified.")
+        scmap = symmap_to_varmap(rn, scmap)
+        scmap = Dict(ModelingToolkit.value(k) => v for (k, v) in scmap)
+        specs = [substitute(s, scmap) for s in specs]
+    end
+
+    vtype = eltype(specs) <: Symbolics.BasicSymbolic ? Num : eltype(specs)
+    Φ = Vector{vtype}()
+    rcmap = reactioncomplexmap(rn)
+    for comp in keys(reactioncomplexmap(rn))
+        subs = map(ce -> getfield(ce, :speciesid), comp)
+        stoich = map(ce -> getfield(ce, :speciesstoich), comp)
+        push!(Φ, prod(vtype[specs[s]^α for (s, α) in zip(subs, stoich)]))
```

Comment:
Isn't this resulting in a different rate law then Catalyst usually generates for ODEs? Where are the factorial factors getting put here? If not in the vector then they need to be in the rate `k` right?

Basically this decomposition should probably support the `combinatoric_ratelaws` keyword and have it `true` by default.